### PR TITLE
CMake improvements

### PR DESCRIPTION
**Pull request by Redix**
- Use static CRT on windows with MSVC
- Use multiple cores with MSVC
- Disable some warnings on win64
- Minor fixes

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -410,6 +411,8 @@ set(TARGETS ${TARGETS_OWN} ${TARGETS_DEP})
 
 foreach(target ${TARGETS})
   if(MSVC)
+    target_compile_options(${target} PRIVATE $<$<CONFIG:Release>:/MT>$<$<CONFIG:Debug>:/MTd>) # Use static CRT
```
Comment (heinrich5991) [inline CMakeLists.txt]: There are a couple of build types, I think both `Debug` and `RelWithDebugInfo` should get debug info. Could you do that and use the negation of `Debug` and `RelWithDebugInfo` for not getting debug info?

(Also, a space between these two generator expressions wouldn't hurt. :) )

File: CMakeLists.txt [hunk 2]
```diff
@@ -424,6 +427,9 @@ endforeach()
 foreach(target ${TARGETS_OWN})
   if(MSVC)
     target_compile_options(${target} PRIVATE /wd4244) # Possible loss of data (float -> int, int -> float, etc.).
+    if(TARGET_BITS EQUAL 64)
+      target_compile_options(${target} PRIVATE /wd4267) # Possible loss of data (size_t - int on win64).
+    endif()
```
Comment (heinrich5991) [inline CMakeLists.txt]: This warning doesn't sound too bad. Can you give an example of where it appears?

Comment (heinrich5991) [inline CMakeLists.txt]: OK, compiling with a similar warning turned on gives a lot of output with `clang`, so I guess we want this to be disabled. Can you disable it unconditionally, independent of 32/64 bit?

**Comment by Redix** [comment]
Fixed the commits

File: CMakeLists.txt (hunk 1 above)
Comment (Redix) [inline CMakeLists.txt]: Done :)

**Comment by heinrich5991** [comment]
Thanks!

**Pull request merged**